internal/cli/command/controlplane/debug.go

```diff
@@ -0,0 +1,241 @@
+
+	return cmd
+}
+
+type debugMetadata struct {
+	Timestamp       time.Time
+	Host            host.InfoStat
+	CLIVersion      string
+	WorkspacePath   string
+	Workspaces      []string
+	InstallerImages []string
+	DockerVersion   string
+}
+
+func runDebug(options debugOptions, banzaiCli cli.Cli) error {
+	logBuffer := new(bytes.Buffer)
+	log.SetOutput(io.MultiWriter(os.Stderr, logBuffer))
+
+	if err := options.Init(); err != nil {
+		return err
+	}
+
+	path := options.filepath()
+	f, err := os.Create(path)
+	if err != nil {
+		return errors.WrapIff(err, "failed to create archive at %q", path)
+	}
+	defer f.Close()
+
+	gzWriter := gzip.NewWriter(f)
+	defer gzWriter.Close()
+
+	tarWriter := tar.NewWriter(gzWriter)
+	defer tarWriter.Close()
+
+	addDir(tarWriter, "")
+
+	meta := debugMetadata{
+		Timestamp:       time.Now(),
+		CLIVersion:      banzaiCli.Version(),
+		WorkspacePath:   options.workspace,
+		Workspaces:      strings.Split(simpleCommand("ls ~/.banzai/pipeline"), "\n"),
+		InstallerImages: strings.Split(simpleCommand("docker image list|grep pipeline-installer"), "\n"),
+		DockerVersion:   simpleCommand("docker --version"),
+	}
+
+	if info, err := host.Info(); err == nil {
+		meta.Host = *info
+	}
+
+	bytes, _ := yaml.Marshal(meta)
+	logResult("add meta.yaml", addFile(tarWriter, "meta.yaml", string(bytes)))
+
+	logResult("create pipeline/", addDir(tarWriter, "pipeline"))
+	logResult("add values.yaml", copyFile(tarWriter, "pipeline/values.yaml", options.valuesPath()))
+
+	// run some terraform diagnostics commands to catch their output in the logs folder
+	var values map[string]interface{}
+	logResult("read values", options.readValues(&values))
+	_, env, err := getImageMetadata(options.cpContext, values, true)
+	logResult("get image metadata", err)
+	logResult("run tf plan", runTerraform("plan", options.cpContext, env))
+	logResult("run tf graph", runTerraform("graph", options.cpContext, env))
+	logResult("run tf state list", runTerraform("state list", options.cpContext, env))
+
+	logResult("create pipeline/installer-logs", addDir(tarWriter, "pipeline/installer-logs"))
+	logDir, logFiles, err := options.listLogs()
+	if err != nil {
+		log.Errorf("listing log files failed: %v", err)
+	} else {
+		for _, file := range logFiles {
+			logResult("add log file", copyFile(tarWriter, filepath.Join("pipeline/installer-logs", file), filepath.Join(logDir, file)))
+		}
+	}
+
+	logResult("add pipeline/files.txt", addFile(tarWriter, "pipeline/files.txt", simpleCommand("find", options.workspace, "-ls")))
+
+	logResult("create pipeline/resources", addDir(tarWriter, "pipeline/resources"))
+	logResult("add pipeline/resources/all.txt", addFile(tarWriter, "pipeline/resources/all.txt", fetchContainerCommandOutputAndError(options.cpContext, []string{"kubectl", "get", "all", "-A", "-owide"}, env)))
```

Comment:
this would need an extra clock cycle from me to understand. i've never used/seen it in the long form

Reply:
Interesting, I'm the other way around, long flags are self-explanatory to me (or when they aren't, the shorthands are not better in any way at least), while shorthands are a lot more ambiguous, for example -o can be an output format or an output file as well.

My general rule of thumb is to always use the long ones when writing "static code" and only use shorthands while writing "live" commands in while doing something IRL synchronously.

I guess it is a clash of preferences then, so uphold the possibility of opting out of the optional suggestion.

Reply:
I would generally agree with this approach, but there are some frequently used commands that most people always use with short flags, and it can be confusing to write out the less known long flag. For example, you know that the more recent files are at the end of the output of the idiomatic `ls -ltr`, but you'd have to find out the same if you see somewhere `ls --long --time --reverse` (which doesn't even work:). I consider common kubectl arguments the same in a kubernetes focused piece of software.

Reply:
You mean when you see `-owide` you don't understand it as `output format wide`, but some other way which doesn't involve the word output?
I can only see this possible if you know "what to expect" from the flag and not "what it means/what it is supposed to be", which IMO is less direct, but I don't want to convince you, so leaving the shorthands in is fine by me, I will try to not comment it when reviewing your pull requests.

Aside from this, the long noun (for example `ns` vs `namespaces` when to use your example of always issuing commands with either `-n` pr `--namespace`) and separated flag name and value (`"-o", "wide"` vs. `"-owide"`) I think still stands.